pkg/client/clientcmd/client_config.go

```diff
@@ -207,17 +209,22 @@ func canIdentifyUser(config client.Config) bool {
 }
 
 // Namespace implements KubeConfig
-func (config DirectClientConfig) Namespace() (string, error) {
+func (config DirectClientConfig) Namespace() (string, bool, error) {
 	if err := config.ConfirmUsable(); err != nil {
-		return "", err
+		return "", false, err
 	}
 
 	configContext := config.getContext()
 
 	if len(configContext.Namespace) == 0 {
-		return api.NamespaceDefault, nil
+		return api.NamespaceDefault, false, nil
 	}
-	return configContext.Namespace, nil
+
+	overridden := false
+	if config.overrides != nil && config.overrides.Context.Namespace != "" {
```

Comment:
Already had the positive test case. I added a negative one in client_config_test.go